### ISSUE TITLE
Lint Rule: No boolean literal for arguments.

### ISSUE THREAD

**Issue by l-cornelius-dol** [denoland/deno_lint#975]
It would be great to have a lint rule to flag when `true` or `false` is passed to a function [edit] and that argument is not the only argument.

As a solution the problem of a vague true/false value passed to a function, the "boolean trap", I have a habit of creating a self-documenting constant and using that instead. Instead of `redraw(true,true)`, I'll define `const ALL_VIEWS = true, INLINE = true;` and then use `redraw(ALL_VIEWS,INLINE);`. [edit: added another argument].

I'd like to be able to flag when I or a member of our team forgets to do this.

Thanks for considering this.

**Comment by bartlomieju** [comment]
I'm open to adding such rule, but it would be an optional rule (ie. required to enable it manually via a config file). PRs are welcome!

**Comment by CRBl69** [comment]
If nobody is already working on this PR, I'd like to give it a try !

**Comment by bartlomieju** [comment]
@CRBl69 please do!

**Comment by jmj0502** [comment]
@bartlomieju Is this issue still open? If so, I would like to give it a try (I may need some context of the code that should be affected in order to introduce the functionality tho')

**Comment by jmj0502** [comment]
@lawrence-dol Question. Do you want the rule to take effect in the following cases as well?
```typescript
invoke(true, remoteServerURL, true);
write(path, true, true);
```
Basically, in cases where there is a mix between named arguments and boolean literals or only in cases that match the example you previously provided?

**Comment by jmj0502** [comment]
@lawrence-dol Perfect! Many thanks for the clarification. I'm almost done so the PR will probably be submitted soon!

**Comment by jmj0502** [comment]
@lawrence-dol @bartlomieju The PR is already pending for review! Look forward to your comments!

**Comment by jmj0502** [comment]
@lawrence-dol What you mention actually makes sense and I have no problem implementing it. When it comes to not allowing any boolean literals, I do agree because any floating literal seems like a magic value to me (IMO). I can proceed to implement the changes in a few minutes, but it I think is important to narrow down the expected behavior. 

To recap, the rule should point out any boolean literals passed as arguments to a function, even if it is a single one.

So the following examples:
```typescript
execute(true);
redraw(true, true);
executeCommand(true, EXECUTION_MODE.MODE_ONE);
```
Will all be marked as improvement points by the linter. Right?

Also, do you think the provided hint still works with these particular changes? Or do you recommend using other hint?

Look forward to your answer c:

**Comment by jmj0502** [comment]
@lawrence-dol Cool! I'll submit my changes in a few minutes c:

**Comment by jmj0502** [comment]
@lawrence-dol Done! The PR is now updated with the changes. Look forward to any comments!

**Comment by jmj0502** [comment]
@bartlomieju @lawrence-dol there was some issue with the format of the code when checked by the actions. I already fixed and uploaded the changes. 👍

**Comment by jmj0502** [comment]
@bartlomieju @lawrence-dol A fix was applied in order to comply with a `clippy` linting recommendation. Now the PR is truly ready!

**Comment by jmj0502** [comment]
@lawrence-dol @bartlomieju The PR already landed so this issue can be closed! c:
